index.js

```diff
@@ -30,6 +30,22 @@ function attachComments(text, ast, opts) {
   return astComments;
 }
 
+// Cache compiled regexps in the likely case the same pragma is used for many
+// files.
+const reForPragma = {};
+function hasPragma(astComments, pragma) {
+  let re;
+  if (reForPragma[pragma] != null) {
+    re = reForPragma[pragma];
+  } else {
+    // @pragma surrounded by any amount of whitespace on either or both sides
+    re = reForPragma[pragma] = new RegExp(`\\s*@${pragma}\\s*`);
```

Comment:
Your `\s*` does absolutely nothing. It matches -- 0 -- or more whitespaces.

Reply:
You need to escape `pragma`, otherwise people can start adding .* to match literally anything.

Reply:
It is also worth thinking about if `sales@prettierjs.com` should match the `@prettier` pragma. Perhaps we should check how Flow checks for `@flow` and do something similar?